File: psuedocode/activities/problem_4/reverse.cc
#include <iostream>
#include <string>

using namespace std;

string reverse(string const& s){
  
  string rev_s = s;
  int n = s.length();

  for (int i = 0; i < n; i++){
    rev_s[i] = s[n - i - 1];
  }
  return rev_s;
}

int main(){
  string s;

  while(cin >> s){
    cout << reverse(s) << endl;
  }
  return 0;
}

File: psuedocode/activities/problem_4/reverse.py
import sys

def reverse(s):

    rev_s = "" 
    n = len(s)

    for i in range(n):
        rev_s += s[n -i - 1]

    return rev_s

s = sys.stdin.readline().strip()

while s != "":
    print(reverse(s))
    s = sys.stdin.readline().strip()
